homeassistant/components/aws/image_processing.py

```diff
@@ -0,0 +1,527 @@
+"""AWS platform for image_processing component."""
+import io
+import logging
+from pathlib import Path
+import re
+
+from PIL import Image, ImageDraw, UnidentifiedImageError
+import aiobotocore
+
+from homeassistant.components.image_processing import (
+    ATTR_AGE,
+    ATTR_CONFIDENCE,
+    ATTR_GENDER,
+    ATTR_GLASSES,
+    ATTR_MOTION,
+    ATTR_NAME,
+    CONF_CONFIDENCE,
+    CONF_ENTITY_ID,
+    CONF_NAME,
+    CONF_SOURCE,
+    ImageProcessingEntity,
+    ImageProcessingFaceEntity,
+)
+from homeassistant.const import ATTR_ENTITY_ID, CONF_PLATFORM, STATE_UNKNOWN
+from homeassistant.core import split_entity_id
+from homeassistant.helpers import entity_platform
+import homeassistant.util.dt as dt_util
+from homeassistant.util.pil import COLOR_RGB_YELLOW, draw_box
+
+from . import SERVICE_INDEX_FACE_SCHEMA
+from .const import (
+    ATTR_LABELS,
+    ATTR_OBJECTS,
+    CONF_ACCESS_KEY_ID,
+    CONF_COLLECTION_ID,
+    CONF_CREDENTIAL_NAME,
+    CONF_DETECTION_ATTRIBUTES,
+    CONF_IDENTIFY_FACES,
+    CONF_PROFILE_NAME,
+    CONF_REGION,
+    CONF_SAVE_FILE_FOLDER,
+    CONF_SAVE_FILE_TIMESTAMP,
+    CONF_SECRET_ACCESS_KEY,
+    CONF_SERVICE,
+    DATA_SESSIONS,
+    EVENT_FACE_INDEXED,
+    EVENT_LABEL_DETECTED,
+    EVENT_OBJECT_DETECTED,
+    SERVICE_INDEX_FACE,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def get_valid_filename(name):
```

Comment:
Is not async

Reply:
Removed all redundant `async` declarations.